### PR TITLE
(DOCSP-29260): Migrate from KMongo

### PR DESCRIPTION
**Pull request by cbullinger**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-29260
Staging - https://docs-mongodbcom-staging.corp.mongodb.com/kotlin/docsworker-xlarge/docsp-29260-migrate/migrate-kmongo/

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Did you run a spell-check?
- [ ] Did you run a grammar-check?
- [ ] Are all the links working?

### CODE REVIEW

File: source/migrate-kmongo.txt [hunk 1]
```diff
@@ -0,0 +1,514 @@
+.. _kotlin-migrate-kmongo:
+
+===================
+Migrate from KMongo
```
Comment (rozza) [inline source/migrate-kmongo.txt:4]: Is Migrate the right word?

This is more a comparison between the two

File: source/migrate-kmongo.txt [hunk 2]
```diff
@@ -0,0 +1,514 @@
...
+
+Both drivers provide support for serializing and deserializing data objects
+in Kotlin to and from BSON.
+
+.. tabs::
+
+  .. tab::
+      :tabid: {+driver-long+}
+
+      You can serialize data classes in the Kotlin driver using the ``kotlinx.serialization``
```
[the remark this answers is not in the document]
Comment (cbullinger) [reply source/migrate-kmongo.txt:310]: Updated

File: source/faq.txt [hunk 3]
```diff
@@ -25,11 +25,7 @@ developed by the MongoDB team and provides a native API for Kotlin
 applications to connect to MongoDB and work with data. It is implemented
 by wrapping the :driver:`MongoDB Java driver </java-drivers/>`.
```
Comment (rustagir) [inline source/faq.txt:27]: Not related to this PR necessarily, but S: move all FAQ entries to top level, right now all but the first are nested under the first even though they are not subsections

File: source/includes/kmongo-description.rst [hunk 4]
```diff
@@ -0,0 +1,5 @@
+`KMongo <https://litote.org/kmongo/>`__ is a popular community-developed library
+for working with MongoDB from Kotlin applications.
+It is also a wrapper around the Java driver.
+It was created prior to the creation of the official Kotlin driver to serve
+the needs of the Kotlin community. As of July 2023, it has been marked as deprecated.
```
Comment (rustagir) [inline source/includes/kmongo-description.rst:5]: S: move the deprecation notice into an important admontition

File: source/migrate-kmongo.txt [hunk 5]
```diff
@@ -0,0 +1,518 @@
...
+          val jedis = collection.find(lt(Jedi::age.name, 30)).toList()
+
+          // Update a document
+          val filter = Filters.eq(Jedi::name.name, "Luke Skywalker")
+          val update = Updates.set(Jedi::age.name, 20)
+          collection.updateOne(filter, update)
+
+          // Delete a document
+          val filter = Filters.eq(Jedi::name.name, "Luke Skywalker")
+          collection.deleteOne(filter)
```
Comment (rustagir) [inline source/migrate-kmongo.txt:113]: S: remove unneeded indentation

File: source/migrate-kmongo.txt [hunk 6]
```diff
@@ -0,0 +1,518 @@
...
+
+      Aggregation pipelines can be built using the ``aggregate`` method and the 
+      ``pipeline`` function:
```
Comment (rustagir) [inline source/migrate-kmongo.txt:116]:
```suggestion
      You can build aggregation pipelines by using the ``aggregate`` method and the 
      ``pipeline`` function: 
```

Comment (rustagir) [reply source/migrate-kmongo.txt:116]: S: In general, change from passive to active second person where possible

File: source/migrate-kmongo.txt [hunk 7]
```diff
@@ -0,0 +1,518 @@
...
+
+        data class Jedi(val name: String)
+
+        val yoda = col.findOne(Jedi::name eq "Yoda")
+
+        // Compile error (2 is not a String)
+        val error = col.findOne(Jedi::name eq 2)
+
+        // Use property reference with instances
+        val yoda2 = col.findOne(yoda::name regex "Yo.*")
```
Comment (rustagir) [inline source/migrate-kmongo.txt:231]: S: fix indentation

File: source/migrate-kmongo.txt [hunk 8]
```diff
@@ -0,0 +1,518 @@
...
+
+        import org.litote.kmongo.MongoOperator.lt
+        import org.litote.kmongo.MongoOperator.match
+        import org.litote.kmongo.MongoOperator.regex
+        import org.litote.kmongo.MongoOperator.sample
+
+        val yoda = col.findOne("{name: {$regex: 'Yo.*'}}")!! 
+        val luke = col.aggregate<Jedi>("""[ {$match:{age:{$lt : ${yoda.age}}}},
+                                            {$sample:{size:1}}
+                                        ]""").first()
```
Comment (rustagir) [inline source/migrate-kmongo.txt:246]: S fix indentation

Comment (rustagir) [reply source/migrate-kmongo.txt:246]: check all code blocks

File: source/migrate-kmongo.txt [hunk 9]
```diff
@@ -0,0 +1,518 @@
...
+        val mongoClient = MongoClient.create(uri)
+        val database = mongoClient.getDatabase("test")
+        val collection = database.getCollection<Jedi>("jedi")
+
+        runBlocking {
+
+          // Async operations
+          val jedi =a Jedi("Luke Skywalker", 19)
+          collection.insertOne(jedi)
+        }
```
Comment (rustagir) [inline source/migrate-kmongo.txt:442]: S fix indentation

File: source/migrate-kmongo.txt [hunk 10]
```diff
@@ -0,0 +1,518 @@
...
+        val client = KMongo.createClient()
+        val database = client.getDatabase("test") 
+        val col = database.getCollection<Jedi>() 
+
+        runBlocking {
+          
+          // Async operations
+          col.insertOne(Jedi("Luke Skywalker", 19))
+          val yoda : Jedi? = col.findOne(Jedi::name eq "Yoda")
+        }
```
Comment (rustagir) [inline source/migrate-kmongo.txt:508]: S fix indentation

**Review by rustagir — APPROVED** [review]
approving with some small suggestions that I think would help the refine the page

File: source/includes/kmongo-description.rst [hunk 11]
```diff
@@ -0,0 +1,5 @@
+`KMongo <https://litote.org/kmongo/>`__ is a popular community-developed library
+for working with MongoDB from Kotlin applications.
+It is also a wrapper around the Java driver.
+It was created prior to the creation of the official Kotlin driver to serve
```
[the remark this answers is not in the document]
Comment (cbullinger) [reply source/includes/kmongo-description.rst:4]: Updated

File: source/faq.txt (hunk 3 above)
Comment (cbullinger) [reply source/faq.txt:27]: Weird. Corrected

File: source/migrate-kmongo.txt (hunk 6 above)
Comment (cbullinger) [reply source/migrate-kmongo.txt:116]: Updated

File: source/migrate-kmongo.txt (hunk 5 above)
Comment (cbullinger) [reply source/migrate-kmongo.txt:113]: Removed

File: source/migrate-kmongo.txt (hunk 8 above)
Comment (cbullinger) [reply source/migrate-kmongo.txt:246]: Fixed

File: source/migrate-kmongo.txt (hunk 7 above)
Comment (cbullinger) [reply source/migrate-kmongo.txt:231]: Fixed

File: source/migrate-kmongo.txt (hunk 9 above)
Comment (cbullinger) [reply source/migrate-kmongo.txt:442]: Fixed

File: source/migrate-kmongo.txt (hunk 10 above)
Comment (cbullinger) [reply source/migrate-kmongo.txt:508]: Fixed

**Pull request merged**